src/AllowList.php

```diff
@@ -5,92 +5,48 @@
 namespace Laminas\Filter;
 
 use Laminas\Stdlib\ArrayUtils;
-use Traversable;
 
+use function array_values;
 use function in_array;
-use function is_array;
 
 /**
  * @psalm-type Options = array{
  *     strict?: bool,
- *     list?: array,
- *     ...
+ *     list?: iterable<array-key, mixed>,
  * }
- * @extends AbstractFilter<Options>
+ * @implements FilterInterface<null>
```

Comment:
I am not sure why there is `null` but yet do not really have the mood to give this a more closer look.
I guess you know why and thus I trust this.